### PR TITLE
Update tools\index.md

### PR DESCRIPTION
**Pull request by Craig-Elre**
Updated tool page with more links, and more explicit information on some reach command-line commands, also switched the commands to the expected default ./reach commands

### CODE REVIEW

File: docs/src/tool/index.md [hunk 1]
```diff
@@ -105,24 +109,30 @@ named `default`.
...
 
 ```cmd
-$ reach init
+$ ./reach init
 ```
 
 # {#ref-usage-run} `reach run`
 
-You can run a simple Reach application by executing
+The `reach run` command with no argumenst starts the `index` application in the current directory by default, but you can set a different directory, application name, or both.
```
Comment (mattaudesse) [inline docs/src/tool/index.md:127]: "argumenst"

File: docs/src/tool/index.md [hunk 2]
```diff
@@ -105,24 +109,30 @@ named `default`.
...
+
+The `reach run` command uses the following syntax:
```
Comment (mattaudesse) [inline docs/src/tool/index.md:129]: I think a purist could prove me wrong but, (programmer-)colloquially-speaking, syntax doesn't feel like the right word here... Maybe "interface" or "exposes these options"?

File: docs/src/tool/index.md [hunk 3]
```diff
@@ -105,24 +109,30 @@ named `default`.
...
 
 ```cmd
-$ reach run [APP or DIR] [ARGS]
+$ ./reach run [APP or DIR] [ARGS]
 ```
 
-`APP` represents a Reach module name without its extension (e.g. "index" by default).
+`APP` represents a Reach module name without its extension (e.g. "index" by default).
```
Comment (mattaudesse) [inline docs/src/tool/index.md:135]: Trailing white-space.

File: docs/src/tool/index.md [hunk 4]
```diff
@@ -226,20 +236,39 @@ may be used in any JavaScript project like any other JavaScript file and library
...
 `reach devnet` supports the following options:
 
-+ `--await-background` --- Run in background and await availability.
-+ The mandatory environment variable `REACH_CONNECTOR_MODE` specifies which devnet to run.
++ `--await-background` --- Run in the background and await availability.
+
+  The mandatory environment variable `REACH_CONNECTOR_MODE` specifies which devnet to run.
 The options are:
 
 + `ETH`, which runs an Ethereum devnet on `localhost:8545`
```
Comment (mattaudesse) [inline docs/src/tool/index.md:260]: These aren't right - `$CONN-live` is used when connecting to existing networks (like testnets or even the mainnet).

https://docs.reach.sh/tool/#p_44

For brevity's sake I think it'd be better to link to a consolidated section of the docs that outlines 'em in one place.

File: docs/src/tool/index.md [hunk 5]
```diff
@@ -292,55 +321,56 @@ development API key), and sets `REACH_RPC_TLS_REJECT_UNVERIFIED` to
...
 
 ```cmd
-$ reach rpc-run python3 -u ./index.py
+$ ./reach rpc-run python3 -u ./index.py
 ```
 
 # {#ref-usage-docker-reset} `reach docker-reset`
 
-You can easily kill and remove all Docker containers by executing
+This command removes all containers and volumes from Docker. If you have any personal containers, those containers will also be removed. Containers can then be restarted in your Docker Images list. It does not uninstall any Docker images.
```
Comment (mattaudesse) [inline docs/src/tool/index.md:329]: Is the "Docker Images list" referring to someplace in Docker Desktop's GUI? At a glance that might be confusing to people who live in the terminal and run everything from there.

File: docs/src/tool/index.md [hunk 6]
```diff
@@ -292,55 +321,56 @@ development API key), and sets `REACH_RPC_TLS_REJECT_UNVERIFIED` to
...
 ```
 
 This can be a useful thing to try if your Docker containers stop responding to requests or otherwise misbehave, or if you have updated your Reach images (with `reach update`) but those changes are not taking effect.
-This command is a loose approximation of "turning Docker off and on again."
 It will affect all Docker containers on your machine, not just those created by Reach.
 
 # {#ref-usage-upgrade} `reach upgrade`
 
-You can upgrade your Reach installation by executing
+This command downloads the most recent `reachsh/reach-cli` Docker image only.
```
Comment (mattaudesse) [inline docs/src/tool/index.md:340]: That's technically not right - it updates the script as well:
https://github.com/reach-sh/reach-lang/blob/f8215673f9d66d5257575b2b392958da18a96c58/reach#L70

...but I think users needn't worry about those details and the original blurb was vague but okay.

(BTW I'm working on a replacement for `reach update` and `reach upgrade` that'll change this entirely.)

File: docs/src/tool/index.md [hunk 7]
```diff
@@ -292,55 +321,56 @@ development API key), and sets `REACH_RPC_TLS_REJECT_UNVERIFIED` to
...
+$ ./reach upgrade
 ```
 
-This may change the default version used by `reach` commands.
+This might change the default version used by `reach` commands.
 
 # {#ref-usage-update} `reach update`
 
-You can update the Docker images used by your Reach installation by executing
+This command downloads and installs the most recent versions of the Reach Docker image. It does not start the Docker images once installed.
```
Comment (mattaudesse) [inline docs/src/tool/index.md:350]: I'd undo this change, personally:

- Docker images aren't really _installed_; they're downloaded and then you've got 'em.

- The bit about not starting is sorta implied since we can't predict what _to_ start.

- This line also has a singular vs. plural mismatch.

File: docs/src/tool/index.md [hunk 8]
```diff
@@ -292,55 +321,56 @@ development API key), and sets `REACH_RPC_TLS_REJECT_UNVERIFIED` to
...
+$ ./reach update
 ```
 
-This may change the patch version used by `reach` commands.
+This might change the patch version used by `reach` commands.
 
 # {#ref-usage-version} `reach version`
 
-You can see what version of Reach you have installed by running
+Check which version of the Reach command-line tool is currently installed. This returns both the semantic version (such as `0.1.7`) and the hashes version (such as [639fa565](https://hub.docker.com/layers/reachsh/reach/639fa565/images/sha256-e72fbb183e559a6f531302843c1d4debb499c9286e0ca4839ae66023c7ba2296?context=explore)). This is helpful when making sure an upgrade to a new version is complete.
```
Comment (mattaudesse) [inline docs/src/tool/index.md:360]: Where are you seeing a hash when you run this?

File: docs/src/tool/index.md [hunk 9]
```diff
@@ -1,6 +1,10 @@
 # {#ref} Tool
 
-This document describes the `reach` tool: how to install it and how to use it.
+This document describes the `reach` tool: how to install it and how to use it. All commands on this page assume that Reach is installed in your project repository, but if it is installed in your `PATH`, remove <b>./</b> in front of the reach commands as in the following:
```
[the remark this answers is not in the document]
Comment (jeapostrophe) [reply docs/src/tool/index.md:3]: And use semantic newlines (each sentence is own its own line, so that diffs look beautiful)

And stylize the last use of `reach` properly

File: docs/src/tool/index.md [hunk 10]
```diff
@@ -65,10 +69,10 @@ In other words, `v0.1.6` is equivalent to `0.1.6`.
 
 # {#ref-usage-compile} `reach compile`
 
-You compile your Reach code by executing
+Compile the Reach code by executing
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:72]: I'd drop the `the`

File: docs/src/tool/index.md [hunk 11]
```diff
@@ -226,20 +236,39 @@ may be used in any JavaScript project like any other JavaScript file and library
...
-You can run a private Reach devnet by executing
+You can run a private Reach devnet by executing the following command:
 
 ```cmd
-$ reach devnet
+$ ./reach devnet
+```
+
+:::note
+If running `reach devnet`, it is recommended to permanently set `REACH_CONNECTOR_MODE` to the desired consensus network. If you did not set this when you originally ran `reach config`, then you can run `reach devnet` with the argument`REACH_CONNECTOR_MODE=[OPTION]` as in the following command.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:246]: Missing space between "argument" and "`"

File: docs/src/tool/index.md (hunk 4 above)
Comment (jeapostrophe) [reply docs/src/tool/index.md:260]: I agree

File: docs/src/tool/index.md [hunk 12]
```diff
@@ -277,10 +306,10 @@ It defaults to `rpc-demo`.
 
 # {#ref-usage-rpc-run} `reach rpc-run`
 
-The sub-command
+The following sub-command
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:309]: I don't think we say "sub-command" elsewhere, so I think we should drop it here

File: docs/src/tool/index.md (hunk 5 above)
Comment (jeapostrophe) [reply docs/src/tool/index.md:329]: I agree.

File: docs/src/tool/index.md (hunk 6 above)
Comment (jeapostrophe) [reply docs/src/tool/index.md:340]: In addition, I don't think we should refer to docker images by name in the documentation... they are internal implementation details. We should describe this generally as "upgrade your tool" (the tool being the thing this document is about)

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/tool/index.md (hunk 8 above)
Comment (Craig-Elre) [reply docs/src/tool/index.md:360]: I didn't this time after you mention this, but when I ran it right before writing this, it displayed both. Not sure why...I will change it.

File: docs/src/tool/index.md [hunk 13]
```diff
@@ -105,24 +109,30 @@ named `default`.
...
+  
+  A `reach compile` usage example is available in [Overview](##over-compile). 
+
+# {#ref-usage-init} `reach init`
 
-## {#ref-usage-init} `reach init`
+Running `reach init` creates the `index.rsh` and `index.mjs` template files required for a basic Reach DApp. This allows you to open the files and start writing code. The `index.rsh` file is the DApp and is written in Reach, and the `index.mjs` file is the frontend of the DApp and is written in JavaScript.
 
-You can create template `index.rsh` and `index.mjs` files for a simple Reach app by running
+To run `reach init`, use the following command:
```
[the remark this answers is not in the document]
Comment (Craig-Elre) [reply docs/src/tool/index.md:119]: It is, but it makes the text a bit more reader-friendly. Honestly, all of them have a bit of redundancy, but the redundancy is used to make it flow better for a reader. I think it is better to keep the second line, but I can take a look at making it less redundant.

File: docs/src/tool/index.md (hunk 2 above)
Comment (Craig-Elre) [reply docs/src/tool/index.md:129]: How about structure?

File: docs/src/tool/index.md (hunk 5 above)
Comment (Craig-Elre) [reply docs/src/tool/index.md:329]: How about this? 

This command removes all containers and volumes from Docker. If you have any personal containers, those containers will also be removed. Containers can then be restarted . It does not remove any Docker images.

For someone like me who uses personal containers, I think it is important for them to know that 1. they will lose their container functionalities after doing this, and 2. this does not require everything to be reconfigured.

File: docs/src/tool/index.md (hunk 2 above)
Comment (mattaudesse) [reply docs/src/tool/index.md:129]: Structure sounds less right than syntax to me. Those aren't terms that programmers/console dorks like me use in this context.

Comment (Craig-Elre) [reply docs/src/tool/index.md:129]: The `reach run` command uses the following interface: Is this proper usage?

File: docs/src/tool/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (Craig-Elre) [reply docs/src/tool/index.md:119]: How about this? 

To create these files, run the following command:

File: docs/src/tool/index.md (hunk 2 above)
Comment (mattaudesse) [reply docs/src/tool/index.md:129]: Yup, I think sounds more natural, but if I'm nitpicking I'd probably say "supports" or "exposes" "_ the following interface".

File: docs/src/tool/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (mattaudesse) [reply docs/src/tool/index.md:119]: Another random observation: I can see now why you made the change from `reach` -> `./reach` based on our Slack discussion but when we were chatting I guess I didn't realize you were talking about modifying it here. (The tutorial is probably a good place for `./reach` though.)

What happens if a beginner tries to run `./reach` from a different directory? `../../reach` and `sub/dir/reach` are just as valid when using something like `reach update`, and we get support questions all the time from people who don't know what `./` means. Like you mention, when `reach` exists in `$PATH` then `./` is unnecessary. I'd recommend leaving `./` off like it was before.

File: docs/src/tool/index.md [hunk 14]
```diff
@@ -1,6 +1,6 @@
 # {#ref} Tool
 
-This document describes the `reach` tool: how to install it and how to use it.
+This document describes the `reach` tool: how to install it and how to use it.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:3]: You accidentally added a trailing space again.

File: docs/src/tool/index.md [hunk 15]
```diff
@@ -105,18 +105,24 @@ named `default`.
 + The environment variable `REACH_DEBUG`, if set to any non-empty value, enables debug messages from the Reach compiler, which will appear in the console.
   This debug information includes: the estimated cost of the contract on Algorand.
   This variable automatically enabled `--intermediate-files`.
+  
+  A `reach compile` usage example is available in [Overview](##over-compile).
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:109]: I believe this indentation makes it go with the list item, but you don't want that

File: docs/src/tool/index.md [hunk 16]
```diff
@@ -226,26 +232,28 @@ may be used in any JavaScript project like any other JavaScript file and library
...
 $ reach devnet
 ```
 
-`reach devnet` supports the following options:
+:::note
+If running `reach devnet`, it is recommended to permanently set `REACH_CONNECTOR_MODE` to the desired consensus network. 
+If you did not set this when you originally ran `reach config`, then you can run `reach devnet` with the argument `REACH_CONNECTOR_MODE=[OPTION]` as in the following command.
+:::
 
-+ `--await-background` --- Run in background and await availability.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:237]: You still want to document this option

File: docs/src/tool/index.md [hunk 17]
```diff
@@ -226,26 +232,28 @@ may be used in any JavaScript project like any other JavaScript file and library
...
-+ The mandatory environment variable `REACH_CONNECTOR_MODE` specifies which devnet to run.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:238]: We still need this too

File: docs/src/tool/index.md [hunk 18]
```diff
@@ -226,26 +232,28 @@ may be used in any JavaScript project like any other JavaScript file and library
...
-The options are:
+```cmd
+$ REACH_CONNECTOR_MODE=ALGO ./reach devnet
+```
 
-+ `ETH`, which runs an Ethereum devnet on `localhost:8545`
-+ `ALGO`, which runs an Algorand devnet on `localhost:4180` and an Algorand indexer on `localhost:8980`
+For more information on devnet options, refer to [Algorand](##ref-network-algo), [Conflux](##ref-network-cfx), or [Ethereum](##ref-network-eth).
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:250]: I'd link to the network page itself, not the individual options. That way we don't forget choices as we add more

File: docs/src/tool/index.md [hunk 19]
```diff
@@ -297,25 +305,24 @@ $ reach rpc-run python3 -u ./index.py
 
 # {#ref-usage-docker-reset} `reach docker-reset`
 
-You can easily kill and remove all Docker containers by executing
+This command removes all containers and volumes from Docker. If you have any personal containers, those containers will also be removed. Containers can then be restarted. It does not remove any Docker images.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:308]: Semantic newline

File: docs/src/tool/index.md [hunk 20]
```diff
@@ -325,19 +332,21 @@ You can update the Docker images used by your Reach installation by executing
 $ reach update
 ```
 
-This may change the patch version used by `reach` commands.
+This might change the patch version used by `reach` commands.
 
 # {#ref-usage-version} `reach version`
 
-You can see what version of Reach you have installed by running
+Check which version of the Reach command-line tool is currently installed. This returns the semantic version (such as `0.1.7`).
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:339]: Newline

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
Sorry for so many nitpicking comments, I feel like it is a sign that we need a style guide to capture these notes for the next person :)

File: docs/src/tool/index.md [hunk 21]
```diff
@@ -76,10 +76,9 @@ and each `EXPORT` is an exported Reach.App.
 
 If no `SOURCE` is provided, then `index.rsh` is used.
 
-If no `EXPORT` is provided, then all the exported Reach.Apps will be compiled. If there are no
-Reach.Apps exported, then the program will be compiled as a library, where its exports are available
-to other Reach programs and frontends. The output name of a library is the same as if it exported a Reach.App
-named `default`.
+If no `EXPORT` is provided, then all the exported Reach.Apps will be compiled.
+If there are no Reach.Apps exported, then the program will be compiled as a library, where its exports are available to other Reach programs and frontends.
+The output name of a library is the same as if it exported a Reach.Appnamed `default`.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:81]: The uses of "Reach.App" here should be "`{!rsh} Reach.App`" and this one lost a space

File: docs/src/tool/index.md [hunk 22]
```diff
@@ -226,26 +231,48 @@ may be used in any JavaScript project like any other JavaScript file and library
...
+$ REACH_CONNECTOR_MODE=ALGO ./reach devnet
+```
+
 `reach devnet` supports the following options:
 
-+ `--await-background` --- Run in background and await availability.
-+ The mandatory environment variable `REACH_CONNECTOR_MODE` specifies which devnet to run.
++ `--await-background` --- Run in the background and await availability.
+
+  The mandatory environment variable `REACH_CONNECTOR_MODE` specifies which devnet to run.
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:253]: The indentation on this makes it appear under the list but we don't want that

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/tool/index.md [hunk 23]
```diff
@@ -33,9 +33,9 @@ Reach has a few sub-commands, each with their own options.
 
 However, all commands support the following options:
 
-+ The environment variable `REACH_VERSION` signifies what version of Reach to use.
++ The environment variable `@{ref("cmd", "REACH_VERSION")}` signifies what version of Reach to use.
 
-Although normally expressed in a [semantic versioning](##guide-versions)-friendly format, e.g. `v0.1` or `v0.1.6`, `REACH_VERSION` also supports:
+Although normally expressed in a [semantic versioning](##guide-versions)-friendly format, e.g. `v0.1` or `v0.1.6`, `@{ref("cmd", "REACH_VERSION")}` also supports:
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:38]: Here's a missing one that needs `{!cmd}`

File: docs/src/tool/index.md [hunk 24]
```diff
@@ -60,28 +60,27 @@ Tip: try entering your desired year in the filter box to skip other tag types.
...
-If no `EXPORT` is provided, then all the exported Reach.Apps will be compiled. If there are no
-Reach.Apps exported, then the program will be compiled as a library, where its exports are available
-to other Reach programs and frontends. The output name of a library is the same as if it exported a Reach.App
-named `default`.
+If no `EXPORT` is provided, then all the exported `{!rsh} Reach.App`s will be compiled.
+If there are no `{!rsh} Reach.App`s exported, then the program will be compiled as a library, where its exports are available to other Reach programs and frontends.
+The output name of a library is the same as if it exported a `{!rsh} Reach.App` named `default`.
 
-`reach compile` supports the following options:
+`@{ref("cmd", "compile")}reach compile` supports the following options:
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:83]: Here

File: docs/src/tool/index.md (hunk 4 above)
Comment (jeapostrophe) [reply docs/src/tool/index.md:256]: Ping on this

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/tool/index.md [hunk 25]
```diff
@@ -226,26 +231,32 @@ may be used in any JavaScript project like any other JavaScript file and library
...
 `reach devnet` supports the following options:
 
-+ `--await-background` --- Run in background and await availability.
-+ The mandatory environment variable `REACH_CONNECTOR_MODE` specifies which devnet to run.
-The options are:
++ `--await-background` --- Run in the background and await availability.
 
-+ `ETH`, which runs an Ethereum devnet on `localhost:8545`
-+ `ALGO`, which runs an Algorand devnet on `localhost:4180` and an Algorand indexer on `localhost:8980`
+For more information on envirionmental variables and `{!cmd} REACH_CONNECTOR_MODE`, refer to (##ref-usage-run).
```
Comment (jeapostrophe) [inline docs/src/tool/index.md:253]: `[Running]`?

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

**Review by jeapostrophe — APPROVED** [review]
